content/frontend-web/react-typescript/migrate-class-component-to-functional-component.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Migrate Class Component to Functional Component
+showMetadata: false
+editable: true
+showToc: true
+order: 3
+---
+
+# Local State
+
+เราสามารถแปลงจาก [Class Component](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/class_components/) ด้านล่างนี้
+
+```ts
+import React from "react";
+
+interface MyProps {
+  message: string;
+};
+interface MyState {
+  count: number; 
+};
+class App extends React.Component<MyProps, MyState> {
+  state: MyState = {
+    // optional second annotation for better type inference
+    count: 0,
+  };
+  render() {
+    return (
+      <div>
+        {this.props.message} {this.state.count}
+      </div>
+    );
+  }
+}
+```
+
+ให้เป็น [Functional Component](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components) แบบนี้ได้ 
+
+```ts
+import React, { useState } from "react";
+
+interface PropTypes {
+  message: string;
+}
+
+const App = ({ message }: PropTypes) => {
+  const [count, setCount] = useState(0);
+
+  return (
+    <div>
+      {message} {count}
+    </div>
+  );
+};
+export default App;
+```
+
+ทำไมถึงไม่แนะนำ `React.FC` หรือ `React.FunctionComponent`/`React.VoidFunctionComponent` อ่านรายละเอียดใน [Functional Component](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components) 

```

Comment:
Good one, btw could you please check if Airbnb suggests us to create a component with `function` keyword instead of `arrow function`?

Reply:
@aaronamm Good catchup, as the [Airbnb](https://airbnb.io/javascript/react/) style guide, they don't mention for functional component yet, they still talk with the only class component. However, most functions are using the `function` keyword. I'll change it to `function` instead. 

Thanks a lot.

Reply:
@mildronize  With function keyword, you can export default right away, e.g.
```ts
export default function ...

```

Reply:
I think they do have suggestion na ja.

![image](https://user-images.githubusercontent.com/2938310/124535520-6d6c0100-de40-11eb-9c6d-5d4d25358b3c.png)

Please check.
https://github.com/airbnb/javascript/tree/master/react?fbclid=IwAR182F3xYrr0ROPJUfsYI8iwxup-Ee_UnvtQPRH94eBSHqqMUybG91iicL4#class-vs-reactcreateclass-vs-stateless

https://www.facebook.com/groups/react.th/posts/3551668064853924

Reply:
@aaronamm Oh, I see, thanks a lot.